src/strings-in-block/class-html.php

```diff
@@ -54,22 +57,24 @@ public function update( \WP_Block_Parser_Block $block, array $string_translation
 
 		if ( $block_queries && isset( $block->innerHTML ) ) {
 
-			$dom   = $this->get_dom( $block->innerHTML );
-			$xpath = new \DOMXPath( $dom );
+			$is_list_block = $this->is_list_block( $block );
```

Comment:
@strategio I think we're missing an abstraction here. The fact that you need to pass `$is_list_block` to several other functions indicates what we need two classes with a common interface. One for "normal" html and one for "list" html.

Could you move the code for `get_inner_HTML` and `set_element_value` into a separate class. Then at this point you instantiate the correct class depending on if it's a list block

```
if( $this->is_list_block ) {
    $block_handler = new ListBlockHandler();
} else 
   $block_handler = new StandardBlockHandler();
}

....
....
list( $text, ) = $block_handler->get_inner_HTML( $element );
....
....
$block_handler->set_element_value( $element, ..... );
```

The names I used are just for example. There are probably better names you can come up with.

Reply:
@brucepearson, I refactored as you suggested (which was not so quick), please check again.